runelite-client/src/main/java/net/runelite/client/game/FreezeManager.java

```diff
@@ -0,0 +1,196 @@
+		{
+			type = FreezeType.ICE_BARRAGE;
+		}
+		else if (graphic == FreezeType.BIND.getGraphic())
+		{
+			type = FreezeType.BIND;
+		}
+		else if (graphic == FreezeType.SNARE.getGraphic())
+		{
+			type = FreezeType.SNARE;
+		}
+		else if (graphic == FreezeType.ENTANGLE.getGraphic())
+		{
+			type = FreezeType.ENTANGLE;
+		}
+
+		if (type == null)
+			return;
+
+		int freezeTicks = type.getTicks();
+
+		if (a instanceof Player)
+		{
+			log.debug("Player freeze queued");
+			OverheadPrayer activeOverhead = OverheadPrayer.fromId(((Player) a).getOverheadIcon());
+
+			if (type.isHalfOnPray() && activeOverhead == OverheadPrayer.PROTECT_MAGIC)
+			{
+				freezeTicks /= 2;
+			}
+
+			a.getFreeze().queueFreeze(type, freezeTicks, false);
+		}
+		else if (a instanceof NPC)
+		{
+			log.debug("NPC freeze queued");
+			//TODO determine halving of ticks when praying
+			a.getFreeze().queueFreeze(type, freezeTicks, false);
+		}
+	}
+
+	private static void processActor(Actor subject)
+	{
+		FreezeInfo freezeInfo = subject.getFreeze();
+		boolean wasImmune = freezeInfo.isImmune();
+
+		freezeInfo.decrementAll();
+
+		//after immunity runs out reset everything (no bars)
+		if (!freezeInfo.isFrozen() && !freezeInfo.isImmune() && freezeInfo.getType() == null && wasImmune)
+		{
+			freezeInfo.resetFreeze();
+			log.debug("{} is no longer immune", subject.getName());
+		}
+
+		//if not frozen, not immune and queued freezes
+		//whilst immune we want to skip this
+		//when we're frozen we cannot have a freeze reapply so skip this too
+		//we must have a freeze queued which can be applied
+		//the freeze type must be null which is implied by us not being in a frozen state
+		if (!freezeInfo.isImmune() && !freezeInfo.isFrozen() && freezeInfo.getType() == null && freezeInfo.hasQueued())
+		{
+			for (FreezeInfo.QueuedFreeze qf : freezeInfo.getQueuedFreezes())
+			{
+				//shouldTrigger will check if the freeze is ready to be activated
+				//the freeze graphic usually occurs 1/2 ticks before the freeze is actually applied
+				//that's why freezes are stored because freezes casted whilst we were immune might take
+				//effect when we're no longer immune.
+				//we must queue all freezes despite one already being queued because in the case where we don't have
+				//pid our queued freeze could potentially inhibit a new freeze from being queued which would actually
+				//end up freezing the subject
+				if (qf.shouldTrigger()) {
+					//the freeze time in ticks cannot simply be derived from the currently activate overhead prayer
+					//and freeze type.
+					//unlike freezes which apply 1/2 ticks after casting the length of the freeze is determined during
+					//the tick the graphic changes, so this information must be stored along with the type
+					int ticks = qf.getLengthTicks();
+					//halving has already been determined on cast
+					freezeInfo.startFreeze(qf.getType(), ticks, subject.getWorldLocation());
```

Comment:
Do you plan to remove the queued freeze in here? It looks like since it is triggered, it is no longer queued or pending, but active.